### PR TITLE
Remove API client-libraries section

### PR DESCRIPTION
**Pull request by onlydole**
Remove the API client-library section in favor of maintained Kubernetes client libraries. 

/assign @zacharysarah 
/assign @jimangel

### CODE REVIEW

**Comment by tengqm** [comment]
The "officially supported" client libraries are already separated from the "community maintained" ones. I am not sure if it is a right move to remove pointers to community projects like these. It smells to me like a step towards more closed community. We are removing all "3rd party" contents, even if the content itself is just one or two sentences rather than any details of the technology or project.
Avoiding "dual hosted" content makes total senses to me, but eliminating all pointers to a broader ecosystem which might help promote the k8s adoption or solve some real life problems is different. So ... my question is whether we are in the right direction and whether we have gone too far if the direction is right.

**Comment by onlydole** [comment]
> The "officially supported" client libraries are already separated from the "community maintained" ones. I am not sure if it is a right move to remove pointers to community projects like these. It smells to me like a step towards more closed community. We are removing all "3rd party" contents, even if the content itself is just one or two sentences rather than any details of the technology or project.
> Avoiding "dual hosted" content makes total senses to me, but eliminating all pointers to a broader ecosystem which might help promote the k8s adoption or solve some real life problems is different. So ... my question is whether we are in the right direction and whether we have gone too far if the direction is right.

Hey @tengqm - might there be a good way we can allow for highlights of these projects in a different context? There have been some discussions about how SIG Docs can responsibly curate projects that are active vs those that are stale. I could see these projects having a better home in one of the "[awesome](https://github.com/sindresorhus/awesome)" lists that exist within the community.

The end goal isn't to be exclusive in **any** way, but instead to be more focused around what the SIGs and upstream community have created and provided. 

I think it's worth a broader conversation and we can discuss further in an upcoming SIG Docs meeting.

**Comment by tengqm** [comment]
@onlydole One way I can think of is a few pages in the reference section. We can list all the out-of-tree provisioners/drivers/libraries there. On such a page, we can provider a disclaimer -- the SIG doesn't actively maintain the lists, the data on these pages are all informational, ... -- whatever statements we want to make.
Also thank you for sharing the "awesome" list. Will check it out.

**Comment by sftim** [comment]
Fix for issue #21563

**Comment by sftim** [comment]
/hold
following a brief discussion in the SIG Docs weekly meeting

By the next weekly meeting I'm hoping more people will have taken a look and either  commented (constructively!) or formed a view to take to the call.

File: content/en/docs/reference/using-api/client-libraries.md [hunk 1]
```diff
@@ -37,44 +37,4 @@ Machinery](https://github.com/kubernetes/community/tree/master/sig-api-machinery
 | JavaScript   | [github.com/kubernetes-client/javascript](https://github.com/kubernetes-client/javascript) | [browse](https://github.com/kubernetes-client/javascript/tree/master/examples)
 | Haskell  | [github.com/kubernetes-client/haskell](https://github.com/kubernetes-client/haskell) | [browse](https://github.com/kubernetes-client/haskell/tree/master/kubernetes-client/example)
 
-
-## Community-maintained client libraries
-
-The following Kubernetes API client libraries are provided and maintained by
-their authors, not the Kubernetes team.
```
Comment (sftim) [inline content/en/docs/reference/using-api/client-libraries.md:44]: How about:
- make this text a Note
- reword it to emphasize a bit more that the links below might be bad / stale / no quality implied

and also (cleanup) sort the list by language, then by project title.

Comment (sftim) [reply content/en/docs/reference/using-api/client-libraries.md:44]: An aside: it would be nice to add a caption to the table.

Comment (onlydole) [reply content/en/docs/reference/using-api/client-libraries.md:44]: I like that, @sftim - I'll adjust accordingly

**Comment by sftim** [comment]
@onlydole is there an issue that this PR is linked to?

**Comment by onlydole** [comment]
We don’t have an issue for this item. I’ll create and link one shortly 👍🏼

**Comment by zacharysarah** [comment]
@sftim #21563 😎

**Comment by kbhawkey** [comment]
Hi @onlydole . Looks like you need to rebase this PR 😄 .
Page preview:
https://deploy-preview-21509--kubernetes-io-master-staging.netlify.app/docs/reference/using-api/client-libraries/

Does it makes sense to move the community links to a separate page (versus removing entirely)? Do we need to create a page for community-supported client libraries in the docs or can we link to a page in ks8/community? We do not actively test these libraries, but get requests to add new client libraries. If you add a community page, do you intend to provide a community link from the `using-api/client-libraries` page?

**Comment by zacharysarah** [comment]
@onlydole 👋 It looks like this PR fell by the wayside. Do you still intend to make this change, modified by @sftim's [review](https://github.com/kubernetes/website/pull/21509#discussion_r439420592)? We need to deal with this content one way or another, and this PR is blocking at least one other (#21112).

File: content/en/docs/reference/using-api/client-libraries.md (hunk 1 above)
Comment (celestehorgan) [reply content/en/docs/reference/using-api/client-libraries.md:44]: Per the 25/8 SIG Docs meeting, we decided the following:
- We'll keep this table, because there are _lots_ of scenarios in which someone might write a Kubernetes-adjacent application or connector and need a non-Go language. Ergo, this table is as a requirement for developing certain kinds of Kubernetes applications.
- We're going to use the new almost-merged 3rd party content warning to provide standardized warning langauage around this.

Follow-up steps for @onlydole, when he re-emerges:

1) Wait for the 3rd party content warning to merge (any day now!)
2) Rebase this branch
3) Use the warning label!

Feel free to ping #sig-docs if you need further clarity on this comment.

**Comment by onlydole** [comment]
Ah! Yes indeed! I will get this buttoned up this week and ping the team as soon as this is taken care of. Thank you for putting this back on my radar!

File: content/en/docs/reference/using-api/client-libraries.md (hunk 1 above)
Comment (onlydole) [reply content/en/docs/reference/using-api/client-libraries.md:44]: Thank you, @celestehorgan!!!

Comment (sftim) [reply content/en/docs/reference/using-api/client-libraries.md:44]: > 3rd party content warning

See PR #23107

**Comment by celestehorgan** [comment]
Added the warning in https://github.com/kubernetes/website/pull/23107, feel free to close @onlydole

**Comment by onlydole** [comment]
You're the best, @celestehorgan! Thank you!!! 🎉
/close
